### ISSUE TITLE
Physical constants

### ISSUE THREAD

**Issue by MilesCranmer** [SymbolicML/DynamicQuantities.jl#26]
We could have a submodule that delivers common physical constants. Already I am finding myself having to type out the speed of light by hand; it would be useful if it were just built-in.

This could work as follows:

```julia
using DynamicQuantities
import DynamicQuantities: Units as U, Constants as C
```

Then, `C` would let you access constants, and `U` the units. This is similar to how `astropy` does it, to avoid mixing namespaces.

e.g., the energy of a 500nm photon

```julia
wavelength = 500*U.nm
energy = C.c*C.h/wavelength
```

This `Constants` namespace could also be exposed within the `@u_str` macro, so you could also write this as

```julia
energy = (1/500) * u"C.c*C.h/nm"
```

---

I'd be interested to know if @odow or @chrisrackauckas have any thoughts or ideas in this direction.

**Comment by odow** [comment]
That seems very reasonable to me.

**Comment by j-fu** [comment]
Hi, how does this square with the physical constants defined in Unitful 
or those in 
https://github.com/JuliaPhysics/PhysicalConstants.jl? 

Codes using these constants from different sources should be guaranteed to work together.  
I think it might make sense to check this in CI...

**Comment by MilesCranmer** [comment]
Could you be more specific, I don't understand your question. PhysicalConstants.jl uses Unitful.jl unfortunately, so we cannot use it directly. The constants in PhysicalConstants.jl are taken from the same source as in #27 - from CODATA 2018. By CI do you mean to check the constants are equal?

**Comment by MilesCranmer** [comment]
I understand, that sounds good to me! Thanks.

**Comment by MilesCranmer** [comment]
Closed with #32
